src/Build/Evaluation/Expander.cs

```diff
@@ -2200,6 +2200,82 @@ internal static ItemTransformFunction GetItemTransformFunction(IElementLocation
                     }
                 }
 
+                /// <summary>
+                /// Intrinsic function that returns all files of a given name that are in the same or ancestor directories of the given items.
+                /// </summary>
+                internal static IEnumerable<Pair<string, S>> GetPathsOfAllFilesAbove(Expander<P, I> expander, IElementLocation elementLocation, bool includeNellEntries, string functionName, IEnumerable<Pair<string, S>> itemsOfType, string[] arguments)
```

Comment:
Should this get special cased in the switch statement, like the other item functions?
eg https://github.com/Microsoft/msbuild/blob/a9780dcd18c79bf93982c886c057de76c8001c95/src/Build/Evaluation/Expander.cs#L3624
To avoid binding cost when you can call straight through.

Reply:
Yes, it should. Excellent catch.

Reply:
I'm taking a look at this now.

Reply:
Actually it isn't clear to me how to do this, as this is an _item_ intrinsic function rather than a "static" intrinsic function like `GetDirectoryNameOfFileAbove`.

Reply:
You're right, I think there's no way to do this for item functions at the moment.